content/en/docs/setup/getting-started/index.md

```diff
@@ -72,142 +52,341 @@ Download the Istio release which includes installation files, samples, and the
+but rather a host name, and the above command will have failed to set the `INGRESS_HOST` environment variable.
+Use the following command to correct the `INGRESS_HOST` value:
+
+{{< text bash >}}
+$ export INGRESS_HOST=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+{{< /text >}}
+
 {{< /warning >}}
 
+**Follow these instructions if your environment does not have an external load balancer and choose a node port instead.**
+
+Set the ingress ports:
+
+{{< text bash >}}
+$ export INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+$ export SECURE_INGRESS_PORT=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+{{< /text >}}
+
+_GKE:_
+
 {{< text bash >}}
-$ kubectl label namespace <namespace> istio-injection=enabled
+$ export INGRESS_HOST=<workerNodeAddress>
 {{< /text >}}
 
-Alternatively, you can manually inject Envoy containers in your application pods before deploying them, using
-[`istioctl kube-inject`](/docs/reference/commands/istioctl/#istioctl-kube-inject):
+You need to create firewall rules to allow the TCP traffic to the `ingressgateway` service's ports.
+Run the following commands to allow the traffic for the HTTP port, the secure port (HTTPS) or both:
 
 {{< text bash >}}
-$ istioctl kube-inject -f <your-app-spec>.yaml | kubectl apply -f -
+$ gcloud compute firewall-rules create allow-gateway-http --allow tcp:$INGRESS_PORT
+$ gcloud compute firewall-rules create allow-gateway-https --allow tcp:$SECURE_INGRESS_PORT
 {{< /text >}}
 
-## Deploy an application or the Bookinfo sample {#Bookinfo}
+_Docker For Desktop:_
+
+{{< text bash >}}
+$ export INGRESS_HOST=127.0.0.1
+{{< /text >}}
 
-If you have an application ready to go, deploy it:
+_Other environments (e.g., IBM Cloud Private, etc.):_
 
 {{< text bash >}}
-$ kubectl create -n <namespace> -f <your-app-spec>.yaml
+$ export INGRESS_HOST=$(kubectl get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
 {{< /text >}}
 
-Alternatively,
-[deploy the Bookinfo sample](/docs/examples/bookinfo/)
-to evaluate Istio's features for traffic routing,
-fault injection, rate limiting, etc. Then, explore the various
-[Istio tasks](/docs/tasks/) that interest you.
+{{< /tab >}}
+
+{{< /tabset >}}
+
+1.  Set `GATEWAY_URL`:
+
+    {{< text bash >}}
+    $ export GATEWAY_URL=$INGRESS_HOST:$INGRESS_PORT
+    {{< /text >}}
+
+1.  Ensure an IP address and port were successfully assigned to the environment variable:
+
+    {{< text bash >}}
+    $ echo $GATEWAY_URL
+    192.168.99.100:32194
+    {{< /text >}}
 
-## Next evaluation steps
+### Verify external access {#confirm}
+
+Confirm that the Bookinfo application is accessible from outside the cluster by
+navigating in your browser to:
+
+{{< text plain >}}
+`http://<your $GATEWAY_URL value>/productpage`
```

Comment:
This URL needs to be wrapped in back-ticks or a text section, but not both. Pick one or the other. Personally, I think back-ticks are better since the features of a text section (e.g., copy button) aren't really that useful here, since most of the string (`<your $GATEWAY_URL value>`) needs to be replaced before using it anyway.

Reply:
And I had previously asked him to put in a text box, so I'll let Adam decide on which way to go with this. I also did see in the Learning series of pages, where they have an _echo_ in a text bash box to actually output the URL (it'll expand the envvars), and have you cut/paste from that output to the browser.

Reply:
Tough call whether or not the echo approach is too basic-tutorial like for the setup instructions, but it is a simple way to get the right value for the browser.

This does get me to wonder if starting the bookinfo app in the setup getting started page is messing up the flow for other docs. The bookinfo example points to the setup instructions as a before you begin step, and then proceeds to give the instructions to start the sample. If the user followed the get started setup instructions, instead of other more advanced install instructions, the following bookinfo instructions are not needed anymore, since bookinfo is already up and running. I think get started and the bookinfo example need to add some text to explain that the get started install instructions are a quick start including bookinfo, so the bookinfo sample instructions are only for users that didn't use the quick start. Otherwise people will be confused about what instructions they need to follow.

Reply:
I have that same concern and noted it somewhere earlier (can't find it right now). We point to getting started in some docs and I don't think they expect that Bookinfo is necessarily installed after that is complete.

Reply:
Going with the echo approach as it is the most user friendly and most accurate. 

Regarding the other comments, the Getting Started is getting hard to optimize for its original purpose because so many other docs depend on it for its sub-sections, e.g. other docs linking to its downloading Istio section, and others. Also when another doc links to Getting Started for one of its sub-sections, readers can easily make the mistake of forgetting to go back to the original doc at the immediate conclusion of the given subsection.

A better way to handle this situation is to remake the truly shareable subsections as single-sourced boilerplates instead of so much linking back and forth. This lets you use the instructions in context everywhere it's needed without linking so much, maintaining single-sourced content re-use, but also letting you remove the boilerplate and customize something one-off as needed in the places where the info truly needs to be different.

Reply:
@adammil2000 I don't think there's really that much linking back and forth with getting started (downloading section is the main one I know about). The bigger problem is that lots of docs link generally back to the setup docs, not getting started specifically, to tell people they need to install istio before they can do pretty much anything else. Users can choose from several ways to install istio, but getting started is the doc that stands out as the doc in the setup section to click first. So, I think that what's missing in this new version is a short description at the top of the doc, similar to what was there before, that says that this is a quick start for new users, but if you are already familiar with Istio or interested in installing other configuration profiles or a more advanced deployment model, follow the installing with istioctl instructions instead. The bookinfo sample should also have a tip/warning in the before you begin section that says if you've installed Istio with the getting started (quick start) guide, you will already have bookinfo deployed and can ignore the following instructions.

Reply:
fixed